core/main_test.go

```diff
@@ -97,3 +102,41 @@ func Test_spotEnabledIsAddedByDefault(t *testing.T) {
 		})
 	}
 }
+
+func Test_addDefaultFilterMode(t *testing.T) {
+	tests := []struct {
+		name string
+		cfg  Config
+		want string
+	}{
+		{
+			name: "Missing FilterMode",
+			cfg:  Config{TagFilteringMode: ""},
+			want: "opt-in",
+		},
+		{
+			name: "Opt-in FilterMode",
+			cfg: Config{
+				TagFilteringMode: "opt-in",
+			},
+			want: "opt-in",
+		},
+		{
+			name: "Opt-out FilterMode",
+			cfg: Config{
+				TagFilteringMode: "opt-out",
+			},
+			want: "opt-out",
+		},
```

Comment:
With the previous comment, you could now add the test of an invalid input as well such as `whatever`, and you'd still want `opt-in`.

Reply:
Done